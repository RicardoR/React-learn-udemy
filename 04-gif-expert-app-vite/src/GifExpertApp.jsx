import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {
        setCategories(categoryList => ['New Category', ...categoryList])
    };

    return (
        <>
            <h1> GifExpertApp</h1>
            <button onClick={onAddCategory}>Add category</button>
            <AddCategory/>
            <ol>
                {
                    categories.map((category, index) => <li key={index}>{category}</li>)
                }
            </ol>
        </>
    );
};

