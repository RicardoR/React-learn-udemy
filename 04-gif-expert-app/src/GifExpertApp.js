import { useState } from "react";
import { AddCategory } from './components/AddCategory';

const GifExpertApp = () => {

    const _categories = ['One Punch', 'Samurai x', 'Dragon Ball']
    const [categories, setCategories] = useState(_categories);
        
    return (
      <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr />

        <ol>
          {categories.map((category, idx) => (
            <li key={`${category}-${idx}`}>{category}</li>
          ))}
        </ol>
      </>
    );
};

export default GifExpertApp;