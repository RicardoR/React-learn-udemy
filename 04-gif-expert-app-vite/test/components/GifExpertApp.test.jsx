import {fireEvent, render, screen} from "@testing-library/react";
import {GifExpertApp} from "../../src/GifExpertApp.jsx";

describe('GifExpertApp Ut', () => {
    test('should match with snapshot', () => {
        const {container} = render(<GifExpertApp/>);
        expect(container).toMatchSnapshot();
    });

    test('should set the category to the input', () => {
        render(<GifExpertApp/>);
        const inputValue = 'Dragon Ball';
        const input = screen.getByRole('textbox');

        fireEvent.input(input, {target: {value: inputValue}});
        expect(input.value).toBe(inputValue);
    });

    test('should add the category', () => {
        const inputValue = {target: {value: 'new category'}};
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, inputValue );
        fireEvent.submit( form );

        expect( screen.findByText(inputValue.target.value) ).toBeTruthy();

    });
});
