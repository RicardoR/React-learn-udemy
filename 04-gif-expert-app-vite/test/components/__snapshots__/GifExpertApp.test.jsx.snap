// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Ut should match with snapshot 1`] = `
<div>
  <h1>
     GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <div>
    <h3>
      Dragon Ball
    </h3>
    <h2>
      Loading...
    </h2>
    <div
      class="card-grid"
    />
  </div>
</div>
`;
